File: web_app/src/services/RoomService.js
import axiosInstance from "./axiosConfigs";

const RoomService = {
    getAll(page, size){
        return axiosInstance.get(`Room/getALlRooms/${page}/${size}`)
    },
    getId(id){
        return axiosInstance.get(`Room/findById/${id}`)
    },
    create(body){
        return axiosInstance.post("Room/createRoom", body)
    },
    update(id, body){
        return axiosInstance.put("Room/updateRoom/"+id, body)
    },
    delete(id){
        return axiosInstance.get(`Room/deleteRoom/${id}`)
    }
}
export default RoomService;

File: be_hotel/src/main/java/no/services/room/RoomService.java
package no.services.room;

import no.modal.entity.Room;
import no.modal.request.BookingFilterRequest;
import no.modal.request.RoomRequest;
import org.springframework.data.domain.Page;

public interface RoomService {
    // post
    Room createRoom(RoomRequest request);

    // get
    Page<Room> getAllRooms(int page, int size);
    Room findById(long id);

    //put
    Room update(RoomRequest request, long id);

    // delete
    boolean deleteRoom(long id);

    //exception
    boolean checkRoomAvailability(BookingFilterRequest request);
}
